.circleci/config.yml

```diff
@@ -47,11 +45,36 @@ jobs:
       - run: npm run tsc
 
   deploy:
-    executor: heroku/default
+    executor: my-executor
     steps:
       - checkout
-      - heroku/install
-      - heroku/deploy-via-git
+      - attach_workspace:
+          at: ~/repo
+      - setup_remote_docker
+      - run:
+          name: Build the image
+          command: docker build -t romppane/bulletin-board:$CIRCLE_SHA1 .
+      - run:
+          name: Login to Docker
+          command: docker login -u $DOCKERHUB_LOGIN -p $DOCKERHUB_PASSWORD
+      - run:
+          name: Push to DockerHub
+          command: docker push romppane/bulletin-board:$CIRCLE_SHA1
+      - run:
+          name: Login to registery
+          command: docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com
+      - run:
+          name: Deploy to Heroku
+          command: docker tag romppane/bulletin-board:$CIRCLE_SHA1 registry.heroku.com/$HEROKU_APP_NAME/web
+      - run:
+          name: Release on Heroku
+          command: docker push registry.heroku.com/$HEROKU_APP_NAME/web
+
+  release:
```

Comment:
Imo `release` step should be merged to `deploy` since their meaning is the same and you always want to deploy to heroku after you have created the image so I don't see reason why they should be different steps :P